driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -722,12 +759,30 @@ public boolean onTimeout(Connection connection, long latency, int retryCount) {
             }
 
             Host queriedHost = current;
-            OperationTimedOutException timeoutException = new OperationTimedOutException(connection.address);
+
+            // This cannot be OperationTimedOutException because
```

Comment:
>  I wonder if we should make that clear in the documentation, as it is currently somewhat unclear with regards to that.

Yes I will try to put some comments in the code.

> would it make sense to create a specific implemetation for this case so users can properly check for it?

Either that, or promote ConnectionException to public. I will try to address that in JAVA-1006.